migration.rst

```diff
@@ -0,0 +1,321 @@
+    } else {
+        $response->send();
+    }
+    $kernel->terminate($request, $response);
+
+There are 2 major deviations from the original file. First of all we make
+``$kernel`` globally available. This allows us to use Symfony features inside
+our existing application and gives access to services configured in our
+Symfony application. This helps us prepare our own code to work better
+within the Symfony application when we transition it over or to replace
+outdated and redundant libraries with for example Symfony components.
+
+The legacy bridge, being called after the Symfony request is being handled,
+is responsible for keeping the old bootstrap-logic or, if you used
+separate php files, making sure to figure out which one to load based on
+the request- and response-objects. Essentially whatever happened
+in your current application's front controller needs to happen there,
+excluding things already handled, like autoloading (assuming you have
+migrated it over to composer). The basic outline of this LegacyBridge
+could look somewhat like this:
+
+.. code block:: php
+
+    <?php
+
+    namespace App;
+
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\Response;
+
+    class LegacyBridge
+    {
+        public static function prepareLegacyScript(
+            Request $request,
+            Response $response,
+            string $publicDirectory
+        ) {
+            /*
+             * If Symfony successfully handled the route,
+             * we do not have to do anything.
+             */
+            if ($response->isNotFound() === false) {
+                return;
+            }
+
+            /*
+             * Either map information from Symfony's Request-object
+             * to the existing application logic, e.g. by overwriting
+             * super globals or populating a custom request class from
+             * the existing framework/application or find out which
+             * PHP file to load, e.g. by looking at
+             * $request->getPathInfo().
+             */
+
+            // return the output from your previous application
+        }
+    }
+
+This is the most generic approach you can take, that is likely to work
+no matter what your previous system was. You might have to account for
+certain "quirks", but since your original application is only started
+after Symfony finished handling the request you reduced the chances
+for side effects and any interference. The major downside is, that
+both systems are not well integrated into each other leading to some
+redundancies. For example since the Symfony application is already
+done handling the request you can not take advantage of kernel
+events or utilize Symfony's routing for determining which legacy
+script to call, leading to some redundancies in both applications.
+
+
+Legacy router provider
+......................
+
+<!-- TODO: Use LegacyRouteProvider to fall back to legacy application -->
```

Comment:
A minor comment: in RST files, comments are created putting two dots and one white space (`.. `) at the beginning of the line:

```rst
.. TODO: Use LegacyRouteProvider to fall back to legacy application
```